### PR TITLE
docs: add introduction

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Changes:
- adds mkdocs.yml for integration with fetchai/docs
- adds an architecture component diagram
- adds an entity relationship diagram for primitive entities
- adds an introduction document explaining the basics of ledger-subquery
  - links to external docs

---

image
image
image

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Introduction
+
+The ledger-subquery is a [SubQuery](https://www.subquery.network/)-baesd indexer for the Fetch ledger.
+This indexer provides a [Graphql](https://www.subquery.network/) API for querying for tracked entities.
```
Comment (VijaySharma815) [inline docs/introduction.md:4]: API for querying for tracked entities. -> API for querying tracked entities.

File: docs/assets/entities.puml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+entity Message {
+  +id
+  +typeUrl
+  json
+  ..relations..
+  transaction
+  block
+
+}
```
Comment (VijaySharma815) [inline docs/assets/entities.puml:40]: extra line!

**Comment by bryanchriswhite** [comment]
Thanks @VijaySharma815!

File: docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Introduction
+
+The ledger-subquery is a [SubQuery](https://www.subquery.network/)-baesd indexer for the Fetch ledger.
```
Comment (Jonathansumner) [inline docs/introduction.md:3]:
```suggestion
The ledger-subquery is a [SubQuery](https://www.subquery.network/)-based indexer for the Fetch ledger.
```

File: docs/introduction.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+### Component Diagram
+
+![subquery architecture component diagram legend](./assets/architecture_legend.svg)
+![subquery architecture component diagram](./assets/architecture.svg)
+
+## Querying
+
+The graphql API relies heavily on [postgraphile (as a library)](https://www.graphile.org/postgraphile/usage-library/).
+Postgraphile plugins also play a critical role; in particular, the [connection-filter](https://github.com/graphile-contrib/postgraphile-plugin-connection-filter) and [pg-aggregates](https://github.com/graphile/pg-aggregates) plugins.
+For more information, please reference their documentation:
```
Comment (Jonathansumner) [inline docs/introduction.md:18]:
```suggestion
For more information, please refer to their documentation:
```
maybe?

**Review by Jonathansumner — APPROVED** [review]
Looks good to me! Nice one

**Pull request merged**